src/autobind.cr

```diff
@@ -39,44 +66,46 @@ while arg = ARGV[i += 1]?
     when "true"      then remove_enum_suffix = true
     else                  remove_enum_suffix = value
     end
-  when "--help"
-    STDERR.puts <<-EOF
-    usage : c2cr [--help] [options] <header.h>
-
-    Some available options are:
-        -I<directory>   Adds directory to search path for include files
-        -D<name>        Adds an implicit #define
-
-    In addition, the CFLAGS environment variable will be used, so you may set it
-    up before compilation when search directories, defines, and other options
-    aren't fixed and can be dynamic.
-
-    The following options control how enum constants are cleaned up. By default
-    the value is false (no cleanup), whereas true will remove matching patterns,
-    while a fixed value will remove just that:
-        --remove-enum-prefix[=true,false,<value>]
-        --remove-enum-suffix[=true,false,<value>]
-    EOF
+  when .starts_with? "--lib-name"
+    abort "Only one library name can be specified", UsageExitCode unless lib_name == "LibC"
+    if arg.includes? '='
+      lib_name = arg[11..-1]
+    elsif name = ARGV[1]?
+      lib_name = name
+    else
+      abort "FATAL: library name argument was specified but no value was received.\n#{Help}", UsageExitCode
+    end
+  when .starts_with? "--parent-module"
+    abort "FATAL: only one module name can be specified", UsageExitCode unless mod_name.nil?
+    err_str = "FATAL: module name argument was specified but no value was received."
+    if arg.includes? '='
+      mod_name = arg[16..-1]
+      abort err_str + '\n' + Help, UsageExitCode if mod_name.empty?
+    else
+      mod_name = ARGV[1]? || abort err_str + '\n' + Help, UsageExitCode
+    end
+  when "--help", "-h"
+    STDERR.puts Help
     exit 0
   else
-    abort "Unknown option: #{arg}"
+    abort "FATAL: Unknown option: #{arg}\n#{Help}", UsageExitCode
   end
 end
 
 Clang.default_c_include_directories cflags
 
-unless header
-  abort "fatal : no header to create bindings for."
-end
+abort "FATAL: no header to create bindings for.\n#{Help}", UsageExitCode if header.empty?
 
 parser = Autobind::Parser.new(
   header,
   cflags,
   remove_enum_prefix: remove_enum_prefix,
   remove_enum_suffix: remove_enum_suffix,
+  name: lib_name,
+  module_name: mod_name # <-- possibly nil
 )
 
 parser.parse
 puts parser.libc_output
 check = parser.check
-abort check if check != true
+abort "#{check}" unless check == true
```

Comment:
Is interpoation necessary? Furthermore, I'd prefer to avoid using an additional keyword , `unless`, when there is no syntax advantage (the classic `if` is shorter here).

Reply:
hahaha that was because I changed that, then changed it back. sorry. I was putting the help text at the end of all the `abort` messages, but I realized it wouldn't really be relevant in that one because you didn't use the CLI wrong, something else gave an error.